perun-core/src/main/java/cz/metacentrum/perun/core/impl/modules/attributes/urn_perun_user_attribute_def_def_sshPublicKey.java

```diff
@@ -20,17 +36,180 @@
 	@Override
 	public void checkAttributeSyntax(PerunSessionImpl perunSession, User user, Attribute attribute) throws WrongAttributeValueException {
 		//Null in value is ok here
-		if(attribute.getValue() == null) return;
-		
+		if (attribute.getValue() == null) return;
+
 		//Testing if some ssh key contains new line character
 		List<String> sshKeys = attribute.valueAsList();
-		for(String sshKey: sshKeys) {
-			if(sshKey != null) {
-				if(sshKey.contains("\n")) throw new WrongAttributeValueException(attribute, user, "One of keys in attribute contains new line character. New line character is not allowed here.");
+		for (String sshKey : sshKeys) {
+			if (sshKey != null) {
+				if (sshKey.contains("\n"))
+					throw new WrongAttributeValueException(attribute, user, "One of keys in attribute contains new line character. New line character is not allowed here.");
+				try {
+					validateSSH(sshKey);
+				} catch (Exception e) {
+					throw new WrongAttributeValueException("Invalid SSH key format: " + e.getMessage());
+				}
+			}
+		}
+	}
+
+	/**
+	 * Checks whether is the SSH key in the correct format. Throws exception if validation fails for any reason.
+	 *
+	 * @param sshKey SSH key to be checked
+	 */
+	private void validateSSH(String sshKey) throws Exception {
+		byte[] bytes = null;
+		int[] pos = {0};
+
+		// look for the Base64 encoded part of the line to decode
+		boolean typeFound = false;
+		String sshKeyTypeValue = "";
+		final String BASE64_FORMAT = "^([A-Za-z0-9+/]{4})*([A-Za-z0-9+/]{3}=|[A-Za-z0-9+/]{2}==)?$";
```

Comment:
See how to use the pattern in `urn_perun_group_resource_attribute_def_def_vomsRoles`.
Also, all the final variables should be created outside of the method.